### PR TITLE
Add composing component priority.

### PR DESCRIPTION
**Pull request by mshima**
Add a new priority to improve prompts order.

This will allow to configure (initialize, prompt, configure and compose) every backend generator before client generators.

Related to https://github.com/jhipster/generator-jhipster/issues/26057#issuecomment-2099556702 and fixes https://github.com/jhipster/generator-jhipster/issues/26079. 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/app/generator.spec.ts [hunk 1]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  describe('questions', () => {
+    describe('without answers', () => {
+      before(async () => {
+        await helpers.run(generatorPath).withSkipWritingPriorities();
+      });
+
+      it('should match order', () => {
+        expect(runResult.askedQuestions.map(({ name }) => name)).toMatchInlineSnapshot(`
+[
+  "baseName",
```
Comment (mshima) [inline generators/app/generator.spec.ts:177]: `project-name` generator

File: generators/app/generator.spec.ts [hunk 2]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "applicationType",
```
Comment (mshima) [inline generators/app/generator.spec.ts:178]: `server` generator

File: generators/app/generator.spec.ts [hunk 3]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "packageName",
```
Comment (mshima) [inline generators/app/generator.spec.ts:179]: `java/bootstrap` generator

File: generators/app/generator.spec.ts [hunk 4]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "buildTool",
```
Comment (mshima) [inline generators/app/generator.spec.ts:180]: `java/build-tool` generator

File: generators/app/generator.spec.ts [hunk 5]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "reactive",
+  "authenticationType",
+  "serverTestFrameworks",
+  "databaseType",
+  "prodDatabaseType",
+  "devDatabaseType",
+  "cacheProvider",
+  "enableHibernateCache",
+  "serverSideOptions",
```
Comment (mshima) [inline generators/app/generator.spec.ts:189]: `spring-boot` generator

File: generators/app/generator.spec.ts [hunk 6]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "enableTranslation",
+  "nativeLanguage",
+  "languages",
```
Comment (mshima) [inline generators/app/generator.spec.ts:192]: `languages` generator

File: generators/app/generator.spec.ts [hunk 7]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
+  "clientFramework",
+  "clientTestFrameworks",
+  "withAdminUi",
+  "clientTheme",
```
Comment (mshima) [inline generators/app/generator.spec.ts:196]: `client` generator

File: generators/app/generator.spec.ts [hunk 8]
```diff
@@ -165,4 +165,110 @@ describe(`generator - ${generator}`, () => {
...
     });
   });
+  describe('questions', () => {
+    describe('without answers', () => {
+      before(async () => {
+        await helpers.run(generatorPath).withSkipWritingPriorities();
+      });
+
+      it('should match order', () => {
+        expect(runResult.askedQuestions.map(({ name }) => name)).toMatchInlineSnapshot(`
```
Comment (Tcharl) [inline generators/app/generator.spec.ts:175]: Locally running `npm test`, the `askedQuestion` returns undefined. Any clue?

Comment (mshima) [reply generators/app/generator.spec.ts:175]: `npm ci`?
Requires an updated version of yeoman-test.

**Pull request merged**